_posts/2023-02-01-user-defined-operators.md

```diff
@@ -0,0 +1,237 @@
+---
+layout: post
+published: true
+title: 'Feature Deep-Dive: User-Defined Operators'
+date: '2023-02-01'
+author: Kamil Śliwak
+category: Explainers
+---
+
+The prominent new feature in [Solidity 0.8.18](https://github.com/ethereum/solidity/releases/tag/v0.8.18)
+is the support for defining operators on [user defined value types](https://docs.soliditylang.org/en/latest/types.html#user-defined-value-types) (UDVTs).
+
+A UDVT is a zero-cost abstraction of an elementary value type that creates a completely new type.
+This is similar to creating an alias, but the new type is distinct from the underlying value type
+and all other UDVTs derived from it.
+The ability to use operators is meant to bring the UDVTs closer to being as natural to use as the
+built-in types and promote the use of type-safe interfaces.
+Below, we are going to take a closer look into the details of this new mechanism.
+
+### Built-in operators
+Solidity provides the following operators for its built-in types:
+
+| Category            | Operators                                                                |
+|---------------------|--------------------------------------------------------------------------|
+| Bitwise             | `&`, `\|`, `^`, `<<`, `>>`, `~`                                          |
+| Arithmetic          | `+`, `-`, `*`, `/`, `%`, `**`                                            |
+| Comparison          | `==`, `!=`, `<`, `<=`, `>`, `>=`                                         |
+| Boolean             | `!`, `&&`, `\|\|`                                                        |
+| Assignment          | `=`, `+=`, `-=`, `^=`, `*=`, `/=`, `%=`, `&=`, `\|=`, `^=`, `<<=`, `>>=` |
+| Increment/decrement | `++`, `--`                                                               |
+| Ternary             | `?:`                                                                     |
+| Other               | `delete`, `new`                                                          |
+
+Most of these operators work only with [value types](https://docs.soliditylang.org/en/latest/types.html#value-types).
+
+With the exception of `?:` and `=`, all of these operators are not available on user-defined
+value types, making their use much less convenient than the built-in ones.
```

Comment:
```suggestion
With the exception of `?:` and `=`, all of these operators are not available on UDTVs, 
making their use much less convenient than the built-in ones.
```

Reply:
I'm using them interchangeably for some variety so it was intentional.